### DISCUSSION TITLE
Unresolved attribute reference 'email' for class 'OAuth2PasswordRequestForm'

### DISCUSSION THREAD

**Discussion by david-shiko** [fastapi/fastapi#6953] [Questions] [answered]
Problem:

Unresolved attribute reference 'email' for class 'OAuth2PasswordRequestForm'

The solution:

Make field username Union, EmailStr or str OAuth2PasswordRequestForm in fastapi/security/oauth2.py module.

Context:
# form_data.email not exists, only form_data.username
user = authenticate_user(crud.get_user_by_email(c), form_data.username, form_data.password)

**Comment by tiangolo** [comment, marked as answer]
Sorry, I can't help you if you don't follow the issue template and provide more information.

If you still have problems please create a new issue and follow the template step by step.

Given that, I'm gonna close this one. ☕
